modules/codegen/src/smithy4s/codegen/internals/SmithyBuild.scala

```diff
@@ -45,3 +107,27 @@ private[internals] final case class SmithyBuildMavenRepository(
 private[codegen] object SmithyBuildMavenRepository {
   implicit val codecs: Codec[SmithyBuildMavenRepository] = deriveCodec
 }
+
+private[codegen] sealed trait SmithyBuildPlugin
+
+private[codegen] object SmithyBuildPlugin {
+  def decode(key: String): Decoder[SmithyBuildPlugin] = key match {
+    case "openapi" => Decoder[OpenApi].widen
+    case other =>
+      Decoder.failedWithMessage(
+        s"Plugin $other is not supported by smithy4s. Currently supported plugins: openapi"
+      )
+  }
+
+  case class OpenApi(config: OpenApiConfig) extends SmithyBuildPlugin
+
+  object OpenApi {
+    implicit val decoder: Decoder[OpenApi] = Decoder[JsonObject].emapTry {
+      obj =>
+        Try {
+          val config = OpenApiConfig.fromNode(Node.parse(obj.toJson.noSpaces))
```

Comment:
This certainly works, but I'd rather reconstruct a node by using a Json folder from circe instead of going via the strings.

Reply:
Yeah, I was not really happy about this way of wiring up the smithy `Node` with circe JSON. I'll have a look at the fold logic and see if I can get it to work without going all the way back to the string level.

Reply:
Done! Number conversion was a little awkward but I _think_ this should minimize any errors introduced by unnecessary rounding.